compiler/src/dmd/dsymbolsem.d

```diff
@@ -98,6 +98,29 @@ private uint setMangleOverride(Dsymbol s, const(char)[] sym)
     return 0;
 }
 
+/**
+ * Apply pragma printf/scanf to FuncDeclarations under `s`,
+ * poking through attribute declarations such as `extern(C)`
+ * but not through aggregates or function bodies.
+ *
+ * Params:
+ *    s = symbol to apply
+ *    printf = `true` for printf, `false` for scanf
```

Comment:
Could take a char, bit nicer at the call site

Reply:
I don't consider it nicer, it would add a ternary expression and a new way to call the function wrongly.

Reply:
ok thats reasonable.